### DISCUSSION TITLE
v0.10.0-beta3

### DISCUSSION THREAD

**Discussion by laipz8200** [langgenius/dify#9360] [Releases]
# ✨ What’s New in v0.10.0-beta3? ✨

> [!IMPORTANT]
> This version is for preview only. Please avoid using it in production environments!

> [!NOTE]
> Your feedback is incredibly valuable to us! We encourage you to share your thoughts and discuss any issues using the GitHub Discussions feature. You can find the entry point for Discussions at the bottom of the release page.

## 🚀 Major Updates

- Enhanced Prompt Editor for LLM Node: You can now utilize `Array<String>` and `Array<Number>` types within the Prompt editor. This improvement aims to enhance the flexibility and usability of prompt design.
- New Podcast Generator Tool: We’ve added a tool to demonstrate file generation within the platform. This feature allows you to explore how content like podcasts can be created through automated workflows.

## 🛠️ Bug Fixes

- We’ve addressed several bugs reported by the community. Thank you for your contributions!

## Quick Start

To upgrade to this version, follow these steps:

1. Clone and enter our repository:

```bash
git clone https://github.com/langgenius/dify.git && cd dify
```

2. Switch to the beta branch:

```bash
git checkout 0.10.0-beta3
```

3. Then follow the [Deploy with Docker Compose > Starting Dify](https://docs.dify.ai/getting-started/install-self-hosted/docker-compose#starting-dify).

We hope you find the new features and fixes useful. As always, your feedback and contributions are welcome to help improve Dify!

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.10.0-beta3'>v0.10.0-beta3</a>.</em>

**Comment by lem1272536013** [comment]
This version exists, the session renaming method is invalid, please check whether there is a problem with the ConversationRenameApi api, or the session renaming is invalid caused by other problems, the ConversationRenameApi request response is successful, but the name value returned is the default "New conversation"

**Reply by laipz8200** [reply]
Thank you for your report! Could you please provide a sample request to reproduce this issue? A curl command, or the endpoint you accessed along with the HTTP body, would be really helpful.

**Reply by lem1272536013** [reply]
To be precise, the auto-generated title feature is no longer working。That is, after creating a new conversation, when the first conversation is initiated with the application, the "name" request is automatically initiated, and the following is the crul command of the request:
curl -X POST 'http://<your ip>/v1/conversations/name' \ 
--header 'Authorization: Bearer {api_key}' \
--header 'Content-Type: application/json' \
--data-raw '{ 
"auto_generate": true }'
![0e0b02e300e39452b1296e80ea7ed32](https://github.com/user-attachments/assets/a30e9ce6-b0b0-4661-b91e-90bc4c573516)
Normally, this "name" request should automatically generate a title based on the content of the conversation. But what is confusing is that the value of "name" in the response back is always "New conversation"
![a1cf1cfdd6e3b5fe4e2dc1003c471a7](https://github.com/user-attachments/assets/0fe1539d-2a2d-46da-a5d5-969d01ebec2e)

**Reply by laipz8200** [reply]
We haven't made any changes to this feature, and based on my testing, it seems to be working just fine. You might want to check the API logs to find out why it's not working on your end.

**Reply by lem1272536013** [reply]
> We haven't made any changes to this feature, and based on my testing, it seems to be working just fine. You might want to check the API logs to find out why it's not working on your end.

@laipz8200  I found the problem, it is indeed a problem with dify itself, I just built a simple application with the community version of dify, the model used is chatgpt4o, which can automatically rename sessions, but after I switch to qwen-max, it will not automatically rename sessions, as shown below：
chatgpt4o:
![3007ff5fad9a4a68682e015d5731291](https://github.com/user-attachments/assets/2442b4cf-389d-4044-9bb8-4e7bbb324787)
qwen:
![5f5743cd7fd3889422c36cf31534ee8](https://github.com/user-attachments/assets/78079f3c-a355-4c32-9f97-a6a61cf18490)

**Reply by lem1272536013** [reply]
But the old version 0.6.14 I deployed before is supported, what is the reason for this, is the subsequent version changing this part of the code logic?

**Reply by laipz8200** [reply]
<img width="394" alt="image" src="https://github.com/user-attachments/assets/f8587f14-fca0-4ab6-98e0-fce7d0cbb8af">
<img width="1526" alt="image" src="https://github.com/user-attachments/assets/bfe8e871-6fd2-4bb9-86c2-508deb28dabd">
Here, it runs very well for me. You can test it on our Cloud service to rule out issues such as network and local environment.

**Comment by kurokobo** [comment]
### Image generation by DALL-E 3 seems to be failed:

![image](https://github.com/user-attachments/assets/7a023913-ab81-4ea1-b013-fa802dd3f5b5)

> Run failed: (psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type uuid: "image"
> LINE 3: WHERE tool_files.id = 'image'::UUID
> ^
> 
> [SQL: SELECT tool_files.id, tool_files.user_id, tool_files.tenant_id, tool_files.conversation_id, tool_files.file_key, tool_files.mimetype, tool_files.original_url, tool_files.name, tool_files.size
> FROM tool_files
> WHERE tool_files.id = %(id_1)s::UUID]
> [parameters: {'id_1': 'image'}]
> (Background on this error at: <https://sqlalche.me/e/20/9h9h>)

DSL: [DALL-E 3.txt](https://github.com/user-attachments/files/17450683/DALL-E.3.txt)

**Reply by kurokobo** [reply]
The same DSL works well on **0.10.0-beta2**:
![image](https://github.com/user-attachments/assets/2ef6643f-a2c3-47e4-a5ab-06ee8d3467f1)

**Reply by laipz8200** [reply]
Thank you for the report! It will be fixed in the next version.
